src/main/java/seedu/address/logic/commands/DeleteTransactionCommand.java

```diff
@@ -41,7 +42,10 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Person selected = lastShownList.get(CURRENT_PERSON.getZeroBased());
-        List<Transaction> transactions = selected.getTransactions();
+        List<Transaction> transactions = new ArrayList<>(model.getFilteredTransactionList());
+        if (transactions.isEmpty()) {
+            transactions = selected.getTransactions();
+        }
```

Comment:
i think this is not needed, since when the transaction list is empty, the next if block will just throw the exception

Reply:
I tested without this check, it fails the cases where deletet is called without ever updating the transactionlist via listt or findt. Is that ok? If yes, I'll update the test cases accordingly

Reply:
I thought we were making the user do `listt` or `findt` first before doing `deletet`? Since we weren't taking in an index for the person for `deletet`?

Reply:
yep, ok i've updated it
thanks for clarifying

Reply:
LGTM!